x-pack/plugins/infra/public/hooks/use_alerts_history.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AsApiContract } from '@kbn/actions-plugin/common';
+import { HttpSetup } from '@kbn/core/public';
+import {
+  ALERT_DURATION,
+  ALERT_RULE_UUID,
+  ALERT_START,
+  ALERT_STATUS,
+  ALERT_TIME_RANGE,
+  ValidFeatureId,
+} from '@kbn/rule-data-utils';
+import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+
+interface Props {
+  featureIds: ValidFeatureId[];
+  ruleId: string;
+  dateRange: {
+    from: string;
+    to: string;
+  };
+}
+interface FetchAlertsHistory {
+  totalTriggeredAlerts: number;
+  histogramTriggeredAlerts: Array<{
+    key_as_string: string;
+    key: number;
+    doc_count: number;
+  }>;
+  error?: string;
+  avgTimeToRecoverUS: number;
+}
+
+interface AlertsHistory {
+  isLoadingAlertsHistory: boolean;
+  errorAlertHistory?: string;
+  alertsHistory?: FetchAlertsHistory;
+}
+export function useAlertsHistory({ featureIds, ruleId, dateRange }: Props) {
```

Comment:
Suggestion: It would be a good idea to use react query for new fetch hooks as it does a lot of things itself behind the scene without needing us to take care of it.

Reply:
@maryam-saeidi, I agree, we already discussed it here https://github.com/elastic/kibana/pull/153930#discussion_r1152820987